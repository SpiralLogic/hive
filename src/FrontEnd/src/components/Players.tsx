import '../css/players.css'
import { FunctionComponent, h } from 'preact';
import { Players } from '../domain';
import GameTile from './GameTile';
import Player from './Player';

const Players: FunctionComponent<{ players: Players }> = (props) => {
  const { players } = props;
  return (
    <aside className="players">
      {players.map((player) => (
        <Player name={player.name} id={player.id} show={!!player.tiles.length}>
          {player.tiles.map((tile) => (
            <GameTile key={tile.id} {...tile} />
          ))}
        </Player>
      ))}
    </aside>
  );
};

Players.displayName = 'Players';
export default Players;
