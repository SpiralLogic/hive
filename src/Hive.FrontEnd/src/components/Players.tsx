import '../css/players.css';

import { FunctionComponent } from 'preact';

import { PlayerId } from '../domain';
import GameTile from './GameTile';
import Player from './Player';
import { useGameState } from '../services/signals';

const Players: FunctionComponent<{ currentPlayer: PlayerId }> = (properties) => {
  const { currentPlayer } = properties;
  const { players } = useGameState();
  return (
    <aside className="players" aria-label={'PLayer pieces not yet on the board'}>
      {players.value.map((player) => (
        <Player key={`${player.id}`} name={player.name} id={player.id}>
          {player.tiles
            .map((tile) => ({
              ...tile,
              moves: undefined,
            }))
            .map((tile) => (
              <GameTile currentPlayer={currentPlayer} key={`${tile.id}`} {...tile} />
            ))}
        </Player>
      ))}
    </aside>
  );
};

Players.displayName = 'Players';
export default Players;
