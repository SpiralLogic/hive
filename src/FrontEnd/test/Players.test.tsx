import { h } from 'preact';
import { render, screen } from '@testing-library/preact';
import Players from '../src/components/Players';

describe('playerList Tests', () => {
  const playerList = () => {
    const ant = { id: 1, playerId: 1, creature: 'ant', moves: [{ q: 1, r: 1 }] };
    const fly = { id: 2, playerId: 0, creature: 'fly', moves: [] };

    const players = [
      { id: 1, name: 'Player 1', tiles: [ant, fly, fly] },
      { id: 2, name: 'Player 2', tiles: [ant] },
    ];

    global.window.history.replaceState({}, global.document.title, `/game/33/1`);
    return render(<Players currentPlayer={0} players={players} />).baseElement;
  };

  it('players are rendered', () => {
    playerList();
    expect(screen.getAllByTitle(/^Player [0-9]$/)).toHaveLength(2);
  });

  it('snapshot', () => {
    expect(playerList()).toMatchSnapshot();
  });
});
