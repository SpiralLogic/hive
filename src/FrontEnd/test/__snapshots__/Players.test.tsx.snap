// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerList Tests snapshot 1`] = `
<body>
  <div>
    <aside
      class="players"
    >
      <div
        class="player player1"
        title="Player 1"
      >
        <div
          class="name"
        >
          Player 1
        </div>
        <div
          class="tiles"
        >
          <div
            class="player1 hex tile ant"
            draggable="true"
            tabindex="0"
            title="Player-1 ant"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#ant"
              />
            </svg>
          </div>
          <div
            class="player0 hex tile fly"
            title="Player-0 fly"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#fly"
              />
            </svg>
          </div>
          <div
            class="player0 hex tile fly"
            title="Player-0 fly"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#fly"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="player player2"
        title="Player 2"
      >
        <div
          class="name"
        >
          Player 2
        </div>
        <div
          class="tiles"
        >
          <div
            class="player1 hex tile ant"
            draggable="true"
            tabindex="0"
            title="Player-1 ant"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#ant"
              />
            </svg>
          </div>
        </div>
      </div>
    </aside>
  </div>
</body>
`;

exports[`playerList Tests snapshot 1`] = `
<body>
  <div>
    <aside
      class="players"
    >
      <div
        class="player player1"
        title="Player 1"
      >
        <div
          class="name"
        >
          Player 1
        </div>
        <div
          class="tiles"
        >
          <div
            class="player1 hex tile ant"
            draggable="true"
            tabindex="0"
            title="Player-1 ant"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#ant"
              />
            </svg>
          </div>
          <div
            class="player0 hex tile fly"
            title="Player-0 fly"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#fly"
              />
            </svg>
          </div>
          <div
            class="player0 hex tile fly"
            title="Player-0 fly"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#fly"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="player player2"
        title="Player 2"
      >
        <div
          class="name"
        >
          Player 2
        </div>
        <div
          class="tiles"
        >
          <div
            class="player1 hex tile ant"
            draggable="true"
            tabindex="0"
            title="Player-1 ant"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#ant"
              />
            </svg>
          </div>
        </div>
      </div>
    </aside>
  </div>
</body>
`;
